bin/logstash

```diff
@@ -58,14 +58,9 @@ if [ "$1" = "-V" ] || [ "$1" = "--version" ]; then
   fi
   echo "logstash $LOGSTASH_VERSION"
 else
-  function classpath() {
-    echo -n "$1"
-    shift
-    while [ $# -gt 0 ] ; do
-      echo -n ":${1}"
-      shift
-    done
-  }
-  CLASSPATH="$(classpath ${LOGSTASH_HOME}/logstash-core/lib/jars/*.jar)"
+  unset CLASSPATH
+  for J in $(cd "${LOGSTASH_JARS}"; ls *.jar); do
+    CLASSPATH=${LOGSTASH_JARS}/${J}:${CLASSPATH}
```

Comment:
This reverses the order of the classpath, _prepending_ each match where the former algorithm _appended_. I'm not sure if this will make a difference (in an ideal world, it _shouldn't_).


It also will always emit an extra trailing colon, since `CLASSPATH` is unset in the first iteration.

We can use standard bash parameter expansion's `:+` operator to only join on colon if `CLASSPATH` is already set:

> `${parameter:+word}`
>  - If parameter is null or unset, nothing is substituted, otherwise the expansion of word is substituted.
> 
> -- [Bash Shell Parameter Expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html)

Modified to emit in the same order as pre-patch, we get:

~~~ bash
unset CLASSPATH
for J in $(cd "${LOGSTASH_JARS}"; ls *.jar); do
  CLASSPATH="${CLASSPATH}${CLASSPATH:+:}${LOGSTASH_JARS}/${J}"
done
~~~

Reply:
> This reverses the order of the classpath, prepending each match where the former algorithm appended. I'm not sure if this will make a difference (in an ideal world, it shouldn't).

Good observation. The jar order should not be important and there was certainly not any expected meaningful  order from the  original `*.jar` parameter. But we can obviously easily reverse the order too for the sake of minimizing changes here.

> It also will always emit an extra trailing colon, since CLASSPATH is unset in the first iteration.

Good catch. 

> We can use standard bash parameter expansion's :+ operator to only join on colon if CLASSPATH is already set:

👍  will do.